app/models/match_events/base.rb

```diff
@@ -14,20 +14,27 @@ def to_partial_path
       'match_events/match_event'
     end
 
+    # There are some events - such as parking, that are not linked to matches/etc.
     belongs_to :match,
       class_name: 'ClientOpportunityMatch',
-      inverse_of: :events
+      inverse_of: :events,
+      required: false
```

Comment:
I believe, though I could be wrong, this should be `optional: true`.   Same goes for all of these `belongs_to` associations.

Reply:
I tried optional: true.  The compiler barfs.  As far as I can tell from documentation optional is a Rails 5 addition.  I would rather leave as is, As it can get it to work with required: false, and was unable to get it to work with optional: true.

Reply:
Sound reasonable.